engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala

```diff
@@ -221,11 +240,14 @@ class RuntimeAsyncCommandsTest
     )
 
     // wait for program to start
-    var isProgramStarted = false
-    var iteration        = 0
+    var isProgramStarted  = false
+    var iteration         = 0
+    var out: List[String] = Nil
     while (!isProgramStarted && iteration < 100) {
-      val out = context.consumeOut
-      Thread.sleep(100)
+      context.out.monitor.synchronized {
+        context.out.monitor.wait(1000)
+        out = context.consumeOut
+      }
       isProgramStarted = out == List("started")
```

Comment:
This code is repeated all over the place. Consider adding
```java
MonitoredByteArrayOutputStream.awaitText(String txt)
```
or similar helper method and call it from the test methods.